lint-rules.md

```diff
@@ -0,0 +1,1280 @@
+        String twopass_mode = "Basic"
+        Boolean use_all_cores = false
+        Float out_filter_mismatch_n_over_L_max = 0.3
+        Float out_filter_mismatch_n_over_read_L_max = 1.0
+        Float out_filter_score_min_over_L_read = 0.66
+        Float out_filter_match_n_min_over_L_read = 0.66
+        Float score_genomic_length_log2_scale = -0.25
+        Float seed_search_start_L_max_over_L_read = 1.0
+        Float align_spliced_mate_map_L_min_over_L_mate = 0.66
+        Float pe_overlap_MMp = 0.01
+        Int run_RNG_seed = 777
+        Int sjdb_score = 2
+        Int read_map_number = -1
+        Int read_quality_score_base = 33
+        Int limit_out_SJ_one_read = 1000
+        Int limit_out_SJ_collapsed = 1000000
+        Int limit_sjdb_insert_n_sj = 1000000
+        Int out_QS_conversion_add = 0
+        Int out_SAM_attr_IH_start = 1
+        Int out_SAM_mapq_unique = 255
+        Int out_SAM_flag_OR = 0
+        Int out_SAM_flag_AND = 65535
+        Int out_filter_multimap_score_range = 1
+        Int out_filter_multimap_n_max = 20
+        Int out_filter_mismatch_n_max = 10
+        Int out_filter_score_min = 0
+        Int out_filter_match_n_min = 0
+        Int score_gap = 0
+        Int score_gap_noncan = -8
+        Int score_gap_GCAG = -4
+        Int score_gap_ATAC = -8
+        Int score_del_open = -2
+        Int score_del_base = -2
+        Int score_ins_open = -2
+        Int score_ins_base = -2
+        Int score_stitch_SJ_shift = 1
+        Int seed_search_start_L_max = 50
+        Int seed_search_L_max = 0
+        Int seed_multimap_n_max = 10000
+        Int seed_per_read_n_max = 1000
+        Int seed_per_window_n_max = 50
+        Int seed_none_loci_per_window = 10
+        Int seed_split_min = 12
+        Int seed_map_min = 5
+        Int align_intron_min = 21
+        Int align_intron_max = 500000
+        Int align_mates_gap_max = 1000000
+        Int align_SJ_overhang_min = 5
+        Int align_sjdb_overhang_min = 1
+        Int align_spliced_mate_map_L_min = 0
+        Int align_windows_per_read_n_max = 10000
+        Int align_transcripts_per_window_n_max = 100
+        Int align_transcripts_per_read_n_max = 10000
+        Int pe_overlap_n_bases_min = 0
+        Int win_anchor_multimap_n_max = 50
+        Int win_bin_n_bits = 16
+        Int win_anchor_dist_n_bins = 9
+        Int win_flank_n_bins = 4
+        Int chim_segment_min = 0
+        Int chim_score_min = 0
+        Int chim_score_drop_max = 20
+        Int chim_score_separation = 10
+        Int chim_score_junction_non_GTAG = -1
+        Int chim_junction_overhang_min = 20
+        Int chim_segment_read_gap_max = 0
+        Int chim_main_degment_mult_n_max = 10
+        Int chim_multimap_n_max = 0
+        Int chim_multimap_score_range = 1
+        Int chim_non_chim_score_drop_min = 20
+        Int twopass1_reads_n = -1
+        Int ncpu = 8
+        Int modify_disk_size_gb = 0
+    }
+```
+
+### `disallowed_input_name`
```

Comment:
- (Strong opinion) The second regex is fine (I'll have to change a few of my WDLs!) but the first is way too strict. What if I want an input named `Int indel_length_cutoff`? It would not be allowed by this lint!

Reply:
Wonderful point! You've convinced me to loosen this rule.